app/routes/titiler/algorithms/tree_cover_loss.py

```diff
@@ -0,0 +1,156 @@
+from collections import OrderedDict, namedtuple
+from typing import Optional, Callable
+
+import numpy as np
+from pydantic import ConfigDict
+from fastapi.logger import logger
+from rio_tiler.models import ImageData
+from titiler.core.algorithm import BaseAlgorithm
+
+from app.models.enumerators.titiler import RenderType
+
+class TreeCoverLoss(BaseAlgorithm):
+
+    title: str = "Tree Cover Loss"
+    description: str = "Decode and visualize tree cover loss"
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    start_date: Optional[int] = 2001
+    end_year: Optional[int]  = 2023
+    render_type: RenderType = RenderType.true_color
+    zoom: int = 12
+    tree_cover_density_threshold: Optional[int] = None
+    tree_cover_density_data: Optional[ImageData] = None
+
+    # metadata
+    input_nbands: int = 2
+    output_nbands: int = 4
+    output_dtype: str = "uint8"
+
+    def __call__(self, img: ImageData) -> ImageData:
+        # Read data
+        self.tree_cover_loss_data = img.data[0]
+        self.intensity = img.data[1]
+        self.no_data = img.array.mask[0]
+
+        # Mask by lossyear and tree cover density filters
+        self.mask = self.create_mask()
+
+        # Create encoded or true color RGB and alpha arrays
+        if self.render_type == RenderType.encoded:
+            rgb = self.create_encoded_rgb()
+            alpha = self.create_encoded_alpha()
+        else:   # true color
+            rgb = self.create_true_color_rgb()
+            alpha = self.create_true_color_alpha()
+
+        # Stack RGB and alpha channels
+        data = np.vstack([rgb, alpha[np.newaxis, ...]]).astype(self.output_dtype)
+        data = np.ma.MaskedArray(data, mask=False)
+    
+        return ImageData(data, assets=img.assets, crs=img.crs, bounds=img.bounds)
+
+    def create_mask(self):
+        mask = ~self.no_data
+
+        # TCL goes from 2001 to 2023 by default, only filter if start_year or end_year is specified
+        if self.start_year != 2001:
+            start_mask = self.tree_cover_loss_data >= (self.start_year - 2000)
+            mask &= start_mask
+
+        if self.end_year != 2023:
+            end_mask = self.tree_cover_loss_data <= (self.end_year - 2000)
+            mask &= end_mask
+
+        # Threshold by TCD if specified
+        if self.tree_cover_density_data is not None:
+            if self.tree_cover_density_threshold is not None:
+                density_mask = self.tree_cover_density_data.array[0, :, :] >= self.tree_cover_density_threshold
```

Comment:
I see you're handling the scenario when all tcd is no data differently from how @danscales did it for carbon data (https://github.com/wri/gfw-tile-cache/blob/ee5f8edb47c9441fee8d90d2b9b54eaa154b40e3/app/routes/titiler/algorithms/tree_cover_loss_drivers.py#L71). Should we settle on a consistent behavior?